src/features/register/registerSlice.jsx

```diff
@@ -2,17 +2,15 @@ import { createSlice, createAsyncThunk } from "@reduxjs/toolkit";
 
 const INITIAL_STATE = {
   loading: false,
-  users: [],
+  users: JSON.parse(localStorage.getItem("users")),
   error: "",
   success: false,
 };
 
 export const registeredUser = createAsyncThunk(
   "register/registeredUser",
   async (user) => {
-    const existingUsers = await JSON.parse(
-      localStorage.getItem("users") || JSON.stringify([])
-    );
+    const existingUsers = await JSON.parse(localStorage.getItem("users"));
```

Comment:
```suggestion
    const existingUsers = JSON.parse(localStorage.getItem("users")) || [];
```
This modification ensures that if localStorage.getItem("users") returns null, it will be treated as an empty array ([]). This approach prevents errors when attempting to spread null into an array and allows the application to handle first-time use or cases where no users are stored in localStorage gracefully.